src/Admitto.Infrastructure/Jobs/JobsWorker.cs

```diff
@@ -0,0 +1,350 @@
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Text.Json;
+using Amolenk.Admitto.Application.Common.Abstractions;
+using Amolenk.Admitto.Domain.Entities;
+using Cronos;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Amolenk.Admitto.Infrastructure.Jobs;
+
+public class JobsWorker(
+    IServiceProvider serviceProvider,
+    IOptions<JobsWorkerOptions> options,
+    ILogger<JobsWorker> logger) : BackgroundService
+{
+    private readonly JobsWorkerOptions _options = options.Value;
+    private readonly ConcurrentDictionary<Guid, Task> _runningJobs = new();
+    private readonly SemaphoreSlim _jobExecutionSemaphore = new(options.Value.MaxConcurrentJobs, options.Value.MaxConcurrentJobs);
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        logger.LogInformation("Jobs worker starting with max {MaxConcurrentJobs} concurrent jobs", _options.MaxConcurrentJobs);
+
+        // Reload running jobs state on startup
+        await ReloadRunningJobsAsync(stoppingToken);
+        
+        // Schedule the orphaned jobs cleanup as a recurring job
+        await ScheduleOrphanedJobsCleanupAsync(stoppingToken);
+
+        var scheduledJobsTask = ProcessScheduledJobsAsync(stoppingToken);
+        var pendingJobsTask = ProcessPendingJobsAsync(stoppingToken);
+
+        await Task.WhenAll(scheduledJobsTask, pendingJobsTask);
+    }
+
+    private async Task ProcessScheduledJobsAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                await CheckScheduledJobsAsync(stoppingToken);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error processing scheduled jobs");
+            }
+
+            await Task.Delay(_options.ScheduledJobsCheckInterval, stoppingToken);
+        }
+    }
+
+    private async Task ProcessPendingJobsAsync(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                await CheckPendingJobsAsync(stoppingToken);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error processing pending jobs");
+            }
+
+            await Task.Delay(TimeSpan.FromSeconds(5), stoppingToken);
+        }
+    }
+
+    public async ValueTask<bool> TryExecuteJob(Guid jobId, CancellationToken cancellationToken = default)
```

Comment:
Let's try to remove the polling and make things a bit simpler. The JobsWorker could keep a list of pending jobs in memory (it's already loading pending jobs when starting for the first time). If a job is executed, add it to the db and the list. Keep trying to process a max amount of parallel jobs from the list. The StartJobCommandHandler only needs to add the job to the list. This will give max performance and remove the need to poll for pending  (regular) jobs, thereby reducing latency.